lxd/devices.go

```diff
@@ -603,23 +603,9 @@ func deviceTaskBalance(s *state.State) {
 	}
 
 	// Set the new pinning
-	for ctn, set := range pinning {
-		// Confirm the container didn't just stop
-		if ctn.InitPID() <= 0 {
-			continue
-		}
-
-		sort.Strings(set)
-		cg, err := ctn.CGroup()
-		if err != nil {
-			logger.Error("balance: Unable to get cgroup struct", logger.Ctx{"name": ctn.Name(), "err": err, "value": strings.Join(set, ",")})
-			continue
-		}
-
-		err = cg.SetCpuset(strings.Join(set, ","))
-		if err != nil {
-			logger.Error("balance: Unable to set cpuset", logger.Ctx{"name": ctn.Name(), "err": err, "value": strings.Join(set, ",")})
-		}
+	for inst, set := range pinning {
+		err = inst.SetAffinity(set)
+		logger.Error("Error setting CPU affinity for the instance", logger.Ctx{"name": inst.Name(), "err": err})
```

Comment:
This needs to be in the form:

```go
err := inst.SetAffinity(set)
if err != nil {
    logger.Error("Error setting CPU affinity for the instance", logger.Ctx{"name": inst.Name(), "err": err})
}
```

Also whilst we are at it can you bring the log context into line with other instance related loggers, so use the form:

```
"project": inst.Project().Name, "instance": inst.Name()
```

That way we can use consistent field naming across LXD and will be able to differentiate an instance of the same name in different projects.

Reply:
ah, my bad. Forgot `if` :-)

Thanks! Fixed!